skymarshal/dexserver/web/templates/password.html

```diff
@@ -22,17 +22,25 @@ <h2 class="theme-heading">log in to your account</h2>
     </div>
 
     {{ if .Invalid }}
-      <div id="login-error" class="dex-error-box">
-        invalid {{ $usernamePrompt }} and password.
+      <div class="theme-form-row">
+        <div id="login-error" class="dex-error-box">
+          invalid {{ $usernamePrompt }} and password.
+        </div>
       </div>
     {{ end }}
 
-    <button tabindex="3" id="submit-login" type="submit" class="dex-btn theme-btn-login theme-btn--primary">submit</button>
+    <div class="theme-btn-wrapper">
+      <div class="theme-form-row">
+        <button tabindex="3" id="submit-login" type="submit" class="dex-btn theme-btn-primary">submit</button>
+      </div>
+      {{ if .BackLink }}
+      <div class="theme-form-row">
+        <button tabindex="3" id="cancel-back" type="button" class="dex-btn theme-btn-cancel" onclick="javascript:history.back()">cancel</button>
```

Comment:
@xtremerui could you change the `onclick` listener to `onclick="location.href='/sky/login'"`? When you just go "back" to the previous page and try to select a new provider, it complains about a database error

Reply:
how did you produce that ? It seems working in my local. @aoldershaw

Reply:
hmm for some reason when i run it locally with `docker-compose -f docker-compose.yml -f hack/overrides/ldap.yml up` i see both LDAP and Local are password connectors, which should not be allowed.

Reply:
and yeah the history back seems not working with button click so Dex throws out an error that `connector ID is already set in the request`

Reply:
1. We [generate an `auth_request` id](https://github.com/concourse/dex/blob/05bffc9535a22afaa3817e0a993c58668f2b2b9f/server/handlers.go#L150-L160) as soon as we navigate to `/sky/issuer/auth` 
2. It sets the `connector_id` as soon as you click one of the options and get navigated to `/sky/issuer/auth/ldap`
3. Clicking the back button puts us back at `/sky/issuer/auth`, but still retains the request id
4. Clicking the other provider (`/sky/issuer/auth/ldap`) will reuse the old request id, which fails because [the `connector_id` has already been set](https://github.com/concourse/dex/blob/05bffc9535a22afaa3817e0a993c58668f2b2b9f/server/handlers.go#L218-L237)

Is it possible for us to detect that the user clicked the back button, and force dex to generate a new `auth_request` id?

Reply:
@chenbh I think the way @aoldershaw suggested is forcing dex to regenerate a new `req`. 

When you switching between providers by clicking cancel button, the `req` in url `../sky/issuer/auth/ldap?req=vm3hkmgon5a33jejpww5gsd5j` is changing. Is that what you meant?

Reply:
Yeah the "cancel" button is working as expected, I'm not sure how you would do make that work for the back button though

Reply:
@chenbh pushed a change to handle the history back event so the `req` could be reset